neuralhydrology/modelzoo/transformer.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from typing import Dict
+import numpy as np
+import math
+
+import torch
+import torch.nn as nn
+
+from neuralhydrology.modelzoo.head import get_head
+from neuralhydrology.modelzoo.basemodel import BaseModel
+
+LOGGER = logging.getLogger(__name__)
+
+class Transformer(BaseModel):
+    """Transformer model class, which relies on PyTorch's TransformerEncoder class.
+
+    This class implements the encoder of a transformer network with a regression or probabilistic head. 
+
+    The model configuration is specified in the config file using the following options:
+    -- transformer_embedding_dimension : int representing the dimension of the input embedding space. 
+                                         This must be dividible by the number of self-attention heads (transformer_nheads).
+    -- transformer_positional_encoding_type : choices to "add" or "concatenate" positional encoding to other model inputs.
+    -- transformer_positional_dropout: fraction of dropout applied to the positional encoding.
+    -- seq_length : integer number of timesteps to treat in the input sequence.
+    -- transformer_nheads : number of self-attention heads.
+    -- transformer_dim_feedforward : dimension of the feed-fowrard networks between self-attention heads.
+    -- transformer_dropout: dropout in the feedforward networks between self-attention heads.
+    -- transformer_nlayers: number of stacked self-attention + feedforward layers.
+
+    Parameters
+    ----------
+    cfg : Config
+        The run configuration.
+    """
+
+    def __init__(self, cfg: Dict):
+        super(Transformer, self).__init__(cfg=cfg)
+
+        if cfg.embedding_hiddens:
+            LOGGER.warning("## Warning: Embedding settings are ignored. Use EmbCudaLSTM for embeddings")
```

Comment:
As noted on Skype, this is an old artifact and should be removed. Furthermore, as stated in other places, you could/should use the `cfg.embedding_hiddens` ( + FC class) also for the transformer embedding, to be more flexible as well not duplicate options.

Reply:
The transformer requires the embedding to include dynamic inputs. This is added as an option to FC.